eventsourcing/src/main/java/org/axonframework/eventsourcing/eventstore/jpa/JpaEventStorageEngine.java

```diff
@@ -152,6 +152,35 @@ protected static <T> DomainEventMessage<T> asDomainEventMessage(EventMessage<T>
                : new GenericDomainEventMessage<>(null, event.getIdentifier(), 0L, event, event::getTimestamp);
     }
 
+    /**
+     * Returns a batch of event data as object entries in the event storage with a
+     * greater than the given {@code token}.Size of event is decided by batchSize()
+     *
+     * @param token     Object describing the global index of the last processed event.
+     * @return A batch of event messages as object stored since the given tracking token.It
+     */
+    protected List<Object[]> fetchEvents(GapAwareTrackingToken token) {
```

Comment:
@smcvb  that is right; I just want to change those queries and not deal with token cleaning and parsing events.

The queries(actually fetchEvents method) could be a Functional<> field in `JdbcEventStorageEngine`, so there is no need to create another method like fetchEvents and also extend  `JdbcEventStorageEngine`. Then it will be set by the Builder

Reply:
In the JDBC variant we went for functional interfaces describing every `PreparedStatementCreation`. 
Although we can achieve something similar for JPA, it already does optimizations under the hood for most scenarios. 
Hence, we figured it to be more important for JDBC.

So, although I fully agree with the possibility, I don't think there's enough merit to go that route at this moment, @altuntasfatih.
As such, the chance you've done is fine for the time being.